src/Http/Response/BadRequest.php

```diff
@@ -0,0 +1,44 @@
+<?php declare(strict_types=1);
+/**
+ * File contains the PSR15 ResponseInterface implementation for
+ * Bad Request responses.
+ *
+ * PHP Version 7
+ *
+ * @category PSR15
+ * @package  Http
+ * @author   Xavier Lecours Boucher <xavier.lecours@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ *
+ * @see https://www.php-fig.org/psr/psr-7/
+ * @see https://www.php-fig.org/psr/psr-15/
+ */
+namespace LORIS\Http\Response;
+
+use \LORIS\Http\Response\JsonResponse;
+
+/**
+ * A LORIS Http Response is an implementation of the PSR15 ResponseInterface
+ * to use in LORIS specific for 400 Bad Request.
+ *
+ * @category PSR15
+ * @package  Http
+ * @author   Xavier Lecours Boucher <xavier.lecours@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class BadRequest extends JsonResponse
```

Comment:
Why are BadRequests always JsonResponses? I think it would make more sense for a BadRequest to be a type of response with an error message, and a middleware can convert that message to the appropriate format based on the 'Content-Type' header of the response. (I'm undecided if that should be part of the PageDecorationMiddleware or a different one, but in either case it needs to be done carefully wrt. the ETag middleware to ensure that different content types get different etags..)

Reply:
When the need arise the change should not be too hard to make.
If it is only for html, the template file could use the json_decode(response->getBody())['error'] as a error message value.